singleflight.go

```diff
@@ -25,10 +25,12 @@ type Group[K comparable, V any] struct {
 // comes in, the duplicate caller waits for the original to complete and
 // receives the same results.
 //
-// The context passed to the fn function is a new context which is canceled when
-// contexts from all callers are canceled, so that no caller is expecting the
-// result. If there are multiple callers, context passed to one caller does not
-// effect the execution and returned values of others.
+// The context passed to the fn function is a context that preserves all values
+// from the passed context but is cancelled by the singleflight only when all
+// awaiting caller's contexts are cancelled (no caller is awaiting the result).
+// If there are multiple callers, context passed to one caller does not affect
+// the execution and returned values of others except if the function result is
+// dependent on the context values.
```

Comment:
This might be regarded as a breaking/behavior change. Wondering if this behavior is desired in the general case, or should this be a separate function (something like `DoWithValues`)?

Reply:
I think that this behaviour is desired in general case, even expected. I would not pollute the package api, just relay on the 0 major version to bump the minor one to reflect the behaviour change.